core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,180 +16,194 @@
  * @fileoverview Service to send changes to a topic to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-
-require('domain/topic/topic-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('EditableTopicBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_TOPIC_DATA_URL_TEMPLATE', 'SUBTOPIC_PAGE_EDITOR_DATA_URL_TEMPLATE',
-  'TOPIC_EDITOR_STORY_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService,
-      EDITABLE_TOPIC_DATA_URL_TEMPLATE, SUBTOPIC_PAGE_EDITOR_DATA_URL_TEMPLATE,
-      TOPIC_EDITOR_STORY_URL_TEMPLATE) {
-    var _fetchTopic = function(
-        topicId, successCallback, errorCallback) {
-      var topicDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_TOPIC_DATA_URL_TEMPLATE, {
-          topic_id: topicId
-        });
-
-      $http.get(topicDataUrl).then(function(response) {
-        if (successCallback) {
-          // The response is passed as a dict with 2 fields and not as 2
-          // parameters, because the successCallback is called as the resolve
-          // callback function in $q in fetchTopic(), and according to its
-          // documentation (https://docs.angularjs.org/api/ng/service/$q),
-          // resolve or reject can have only a single parameter.
-          successCallback({
-            topicDict: angular.copy(response.data.topic_dict),
-            groupedSkillSummaries: angular.copy(
-              response.data.grouped_skill_summary_dicts),
-            skillIdToDescriptionDict: angular.copy(
-              response.data.skill_id_to_description_dict),
-            skillIdToRubricsDict: angular.copy(
-              response.data.skill_id_to_rubrics_dict)
-          });
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { TopicDomainConstants } from 'domain/topic/topic-domain.constants';
+import { AppConstants } from 'app.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableTopicBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) { }
```

Comment:
Remove the space between the empty braces.

Reply:
Done